dive_core/dive_command_hierarchy.cpp

```diff
@@ -0,0 +1,204 @@
+
+        for (uint64_t node_index = 1; node_index < num_pm4_nodes; ++node_index)
+        {
+            DIVE_ASSERT(m_pm4_command_hierarchy_creator.get_node_children(topology, 0).size() ==
+                        m_pm4_command_hierarchy_creator.get_node_children(topology, 1).size());
+            cur_topology.AddChildren(node_index,
+                                     m_pm4_command_hierarchy_creator
+                                     .get_node_children(topology, 0)[node_index]);
+            cur_topology.AddSharedChildren(node_index,
+                                           m_pm4_command_hierarchy_creator
+                                           .get_node_children(topology, 1)[node_index]);
+        }
+
+        cur_topology.m_start_shared_child = std::move(
+        m_pm4_command_hierarchy_creator.get_node_start_shared_child(topology));
+        cur_topology.m_end_shared_child = std::move(
+        m_pm4_command_hierarchy_creator.get_node_end_shared_child(topology));
+        cur_topology.m_root_node_index = std::move(
+        m_pm4_command_hierarchy_creator.get_node_root_node_index(topology));
+
+        // Add the gfxr nodes to the topology.
+        if (topology == CommandHierarchy::kAllEventTopology)
+        {
+            // Build the internal GFXR hierarchy for all non-submit nodes.
+            // This includes the self-loop filter to ensure no submit nodes are included.
+            const auto &gfxr_dive_indices = m_gfxr_command_hierarchy_creator
+                                            .GetCreatedDiveIndices();
+            for (uint64_t node_index = num_pm4_nodes; node_index < total_num_nodes; ++node_index)
+            {
+                uint64_t    gfxr_dive_index = gfxr_dive_indices.at(node_index);
+                const auto &children = m_gfxr_command_hierarchy_creator.get_node_children(
+                topology)[gfxr_dive_index];
+
+                DiveVector<uint64_t> filtered_children;
+                for (uint64_t child_node : children)
+                {
+                    if (child_node != node_index)
+                    {
+                        if (m_command_hierarchy.GetNodeType(child_node) !=
+                            NodeType::kGfxrVulkanSubmitNode)
+                        {
+                            filtered_children.push_back(child_node);
+                        }
+                    }
+                }
+                cur_topology.AddChildren(node_index, filtered_children);
+            }
+
+            // Identify ALL GFXR submit nodes.
+            DiveVector<uint64_t> gfxr_submit_nodes;
+            for (uint64_t node_index = num_pm4_nodes; node_index < total_num_nodes; ++node_index)
+            {
+                std::string desc = m_command_hierarchy.GetNodeDesc(node_index);
+                if (m_command_hierarchy.GetNodeType(node_index) == NodeType::kGfxrVulkanSubmitNode)
+                {
+                    std::cout << "SUBMIT INDEX: " << node_index << std::endl;
```

Comment:
out of curiosity, is this only for debug?

Reply:
Removed. Was used for debugging.